src/main/java/com/yurjinia/project_structure/column/service/ColumnService.java

```diff
@@ -82,10 +83,21 @@ public List<ColumnDTO> getColumns(String projectCode, String boardCode) {
                 .toList();
     }
 
+    @Transactional
     public void deleteColumn(String projectCode, String boardCode, String columnName) {
         ColumnEntity columnEntity = getColumnByName(projectCode, boardCode, columnName);
 
+        List<ColumnEntity> columns = boardService.getBoard(boardCode, projectCode).getColumns()
+                .stream()
+                .filter(column -> !column.getId().equals(columnEntity.getId()))
+                .sorted(Comparator.comparing(ColumnEntity::getColumnPosition))
+                .toList();
+
         columnRepository.delete(columnEntity);
+
+        IntStream.range(0, columns.size()).forEach(i -> columns.get(i).setColumnPosition((long) i));
```

Comment:
```java
IntStream.range(0, columns.size()).forEach((long i) -> columns.get(i).setColumnPosition(i));
```

Reply:
Method columns.get() accepts int, not long. So we will anyway cast it to int each iteration